lib/middleware/serveResources.js

```diff
@@ -95,10 +95,18 @@ function createMiddleware({resources, middlewareUtil}) {
 
 			// Enable ETag caching
 			const statInfo = resource.getStatInfo();
-			if (statInfo?.size !== undefined) {
-				res.setHeader("ETag", etag(statInfo));
+			if (statInfo?.size !== undefined && !resource.isModified()) {
+				let etagHeader = etag(statInfo);
+				if (resource.getProject()) {
+					// Add project version to ETag to invalidate cache when project version changes.
+					// This is necessary to invalidate files with ${version} placeholders.
+					etagHeader = etagHeader.replace(/"$/g, `-${resource.getProject().getVersion()}"`);
```

Comment:
```suggestion
					etagHeader = etagHeader.slice(0, -1) + `-${resource.getProject().getVersion()}"`);
```

Maybe no regex required?